pkg/providers/imagefamily/agentbakerbootstrap/provisionclientbootstrap.go

```diff
@@ -0,0 +1,239 @@
+
+func (p ProvisionClientBootstrap) GetCustomDataAndCSE(ctx context.Context) (string, string, error) {
+	if p.IsWindows {
+		// TODO(Windows)
+		return "", "", fmt.Errorf("windows is not supported")
+	}
+
+	labels := lo.Assign(map[string]string{}, p.Labels)
+	getAgentbakerGeneratedLabels(p.ResourceGroup, labels)
+
+	provisionProfile := &models.ProvisionProfile{
+		Name:                     lo.ToPtr(""),
+		VMSize:                   lo.ToPtr(p.InstanceType.Name),
+		Distro:                   lo.ToPtr(p.ImageDistro),
+		CustomNodeLabels:         labels,
+		OrchestratorVersion:      lo.ToPtr(p.KubernetesVersion),
+		VnetSubnetID:             lo.ToPtr(p.SubnetID),
+		StorageProfile:           lo.ToPtr(p.StorageProfile),
+		NodeInitializationTaints: lo.Map(p.StartupTaints, func(taint v1.Taint, _ int) string { return taint.ToString() }),
+		NodeTaints:               lo.Map(p.Taints, func(taint v1.Taint, _ int) string { return taint.ToString() }),
+		SecurityProfile: &models.AgentPoolSecurityProfile{
+			SSHAccess: lo.ToPtr(models.SSHAccess_LocalUser),
+			// EnableVTPM:       lo.ToPtr(false), // Unsupported as of now (Trusted launch)
+			// EnableSecureBoot: lo.ToPtr(false), // Unsupported as of now (Trusted launch)
+		},
+
+		// VnetCidrs:               []string{},                                      // Unsupported as of now; TODO(Windows)
+		// MessageOfTheDay:         lo.ToPtr(""),                                    // Unsupported as of now
+		// AgentPoolWindowsProfile: &models.AgentPoolWindowsProfile{},               // Unsupported as of now; TODO(Windows)
+		// KubeletDiskType:         lo.ToPtr(models.KubeletDiskType_Unspecified),    // Unsupported as of now
+		// CustomLinuxOSConfig:     &models.CustomLinuxOSConfig{},                   // Unsupported as of now (sysctl)
+		// EnableFIPS:              lo.ToPtr(false),                                 // Unsupported as of now
+		// GpuInstanceProfile:      lo.ToPtr(models.GPUInstanceProfile_Unspecified), // Unsupported as of now (MIG)
+		// WorkloadRuntime:         lo.ToPtr(models.WorkloadRuntime_Unspecified),    // Unsupported as of now (Kata)
+		// ArtifactStreamingProfile: &models.ArtifactStreamingProfile{
+		// Enabled: lo.ToPtr(false), // Unsupported as of now
+		// },
+	}
+
+	if p.Arch == "amd64" {
+		provisionProfile.Architecture = lo.ToPtr("x64")
+	} else {
+		provisionProfile.Architecture = lo.ToPtr(p.Arch)
+	}
+
+	switch p.ImageFamily {
+	case v1alpha2.Ubuntu2204ImageFamily:
+		provisionProfile.OsSku = to.Ptr(models.OSSKU_Ubuntu)
+	case v1alpha2.AzureLinuxImageFamily:
+		provisionProfile.OsSku = to.Ptr(models.OSSKU_AzureLinux)
+	default:
+		provisionProfile.OsSku = to.Ptr(models.OSSKU_Ubuntu)
+	}
+
+	if p.KubeletConfig != nil {
+		provisionProfile.CustomKubeletConfig = &models.CustomKubeletConfig{
+			// AllowedUnsafeSysctls: ..., // Unsupported as of now
+			CPUCfsQuota: p.KubeletConfig.CPUCFSQuota,
+		}
+	}
+
```

Comment:
Can you leave some comments here on whats defaulted in NPS vs what is not? Its very confusing to understand what flags we are getting vs need to be passing in.

Reply:
Required fields are marked in the swagger spec.